src/dfx-core/src/config/model/dfinity.rs

```diff
@@ -75,21 +75,20 @@ pub struct ConfigCanistersCanisterRemote {
     pub id: BTreeMap<String, Principal>,
 }
 
+/// # Wasm Optimization Levels
+/// Wasm optimization levels that are passed to `wasm-opt`. "cycles" defaults to O3, "size" defaults to Oz.
+/// O4 through O0 focus on performance (with O0 performing no optimizations), and Oz and Os focus on reducing binary size, where Oz is more aggressive than Os.
+/// O3 and Oz empirically give best cycle savings and code size savings respectively.
 #[derive(Copy, Clone, Debug, PartialEq, Eq, Serialize, Deserialize, JsonSchema)]
 pub enum WasmOptLevel {
-    // Defaults to O3
-    #[serde(rename = "cycles")]
-    Cycles,
-    // Defaults to Oz
-    #[serde(rename = "size")]
-    Size,
-    // Specific performance levels
     O4,
+    #[serde(alias = "cycles")]
```

Comment:
Two concerns with using `serde alias`:
1. Any code that rewrites dfx.json using these types would  read `"cycles"` and write `"O3"`.  We don't have any such code now (ever since we removed `dfx config`), but if we add any in the future we will have to keep this in mind
2. The JSON schema for WasmOptLevel doesn't include the valid enum values of `cycles` and `size`.

My suggestion would be to re-add the `Cycles` and `Size` enum kinds, and transform them to `O3` and `Oz` in one place in the code rather than through serde.